src/cdk-experimental/combobox/combobox.ts

```diff
@@ -6,15 +6,194 @@
+      this._contentType = type;
+    });
+  }
+
+  ngOnDestroy() {
+    this.opened.complete();
+    this.closed.complete();
+    this.panelValueChanged.complete();
+  }
+
+  /** Toggles the open state of the panel. */
+  toggle() {
+    if (this.hasPanel()) {
+      this.isOpen() ? this.close() : this.open();
+    }
+  }
+
+  /** If the combobox is closed and not disabled, opens the panel. */
+  open() {
+    if (!this.isOpen() && !this.disabled) {
+      this.opened.next();
+      this._overlayRef = this._overlayRef || this._overlay.create(this._getOverlayConfig());
+      this._overlayRef.attach(this._getPanelContent());
+    }
+  }
+
+  /** If the combobox is open and not disabled, closes the panel. */
+  close() {
+    if (this.isOpen() && !this.disabled) {
+      this.closed.next();
+      this._overlayRef.detach();
+    }
+  }
+
+  /** Returns true if panel is currently opened. */
+  isOpen(): boolean {
+    return this._overlayRef ? this._overlayRef.hasAttached() : false;
+  }
+
+  /** Returns true if combobox has a child panel. */
+  hasPanel(): boolean {
+    return !!this.comboboxPanel;
+  }
+
+  private _setComboboxValue(value: T) {
+    const valueChanged = (this.value !== value);
+    this.value = value;
+
+    if (valueChanged) {
+      this.panelValueChanged.emit(value);
+    }
+  }
+
+  private _getOverlayConfig() {
+    return new OverlayConfig({
+      positionStrategy: this._getOverlayPositionStrategy(),
+      scrollStrategy: this._overlay.scrollStrategies.block(),
+      direction: this._directionality,
+    });
+  }
+
+  private _getOverlayPositionStrategy(): FlexibleConnectedPositionStrategy {
+    return this._overlay
+        .position()
+        .flexibleConnectedTo(this._elementRef)
+        .withPositions(this._getOverlayPositions());
+  }
+
+  private _getOverlayPositions(): ConnectedPosition[] {
+    return [
+      {originX: 'start', originY: 'bottom', overlayX: 'start', overlayY: 'top'},
+      {originX: 'start', originY: 'top', overlayX: 'start', overlayY: 'bottom'},
+      {originX: 'end', originY: 'bottom', overlayX: 'end', overlayY: 'top'},
+      {originX: 'end', originY: 'top', overlayX: 'end', overlayY: 'bottom'},
+    ];
+  }
+
+  private _getPanelContent() {
+    const hasPanelChanged = this._panel?._templateRef !== this._panelContent?.templateRef;
+    if (this._panel && (!this._panel || hasPanelChanged)) {
+      this._panelContent = new TemplatePortal(this._panel._templateRef, this._viewContainerRef);
+    }
+
+    return this._panelContent;
+  }
+
+  private _coerceOpenActionProperty(input: any): OpenAction[] {
+    const actions: OpenAction[] = [];
+
+    const inputArray = coerceArray(input);
+    for (const action of inputArray) {
+      if (action !== 'focus' && action !== 'click' && action !== 'downKey' && action !== 'toggle') {
+        throw Error('Not a valid open action.');
+      } else {
+        actions.push(action);
+      }
+    }
+
+    return actions;
```

Comment:
I would just have this whole function be:
```suggestion
return input;
```
And add the rest of the behavior when you add support for splitting a string